frontend/public/components/routes.tsx

```diff
@@ -77,20 +77,92 @@ const getRouteLabel = (route) => {
   return label;
 };
 
-export const RouteHostname: React.SFC<RouteHostnameProps> = ({obj}) => <div>
+export const RouteLocation: React.SFC<RouteHostnameProps> = ({obj}) => <div>
   {isWebRoute(obj) ? <a href={getRouteWebURL(obj)} target="_blank" rel="noopener noreferrer">
     {getRouteLabel(obj)}
     <i className="fa fa-external-link" style={{paddingLeft: '4px'}} aria-hidden="true"/>
   </a> : getRouteLabel(obj)
   }
 </div>;
 
+export const RouteStatus: React.SFC<RouteHostnameProps> = ({obj: route}) => {
+  let atLeastOneAdmitted: boolean = false;
+
+  if (!route.status || !route.status.ingress) {
+    return <span>
+      <div className="fa fa-hourglass-half co-m-route-status-icon" aria-hidden="true"></div>
+      Pending
+    </span>;
+  }
+
+  _.each(route.status.ingress, (ingress) => {
```

Comment:
We need to be more defensive here and use `_.get(route, 'status.ingress', []).forEach()` (k8s doesn't always give us all the fields).

Reply:
`_.each` can take null/undefined and status should always be there

Reply:
Hey @alecmerdler just want to understand the reasoning behind the thumbs down. is there a case I missed?

Reply:
Specifically here it can never be a runtime error since route.status is never empty:

https://github.com/openshift/api/blob/master/route/v1/types.go#L37

Reply:
If the aggregated API server is broken, than there won't be a `status` block (this happens with Operators+CRDs). Right now missing fields break the whole app (requires refresh). We should fix that by using React error boundaries, but in the meantime we need to be more defensive.

Reply:
What do you mean by "broken"? Returning incorrect data? If the route API is broken, it's a serious problem generally in the cluster... None of the external URLs for your apps will work.

Does this mean we should always to use `_.get` everywhere for fields that should always be there, for instance `metadata.name` and `metadata.uid`?

Reply:
Generally only use `_.get()` for `spec` and `status`; the other fields are guaranteed by k8s itself. Believe me, I don't like it either (because we lose all the benefits of TypeScript), but I got tired of people complaining to me that the UI was broken when it was really their Operator/controller not writing `status` fields back.

Reply:
OK so I think the Operator+CRD case might be a special one... Routes are serializing this struct:

https://github.com/openshift/api/blob/master/route/v1/types.go#L29-L38

Even if status isn't filled in the property will be there. Note that `metadata` is not guaranteed for all k8s resources. Can we reevaluate if we run into a problem? I've never seen an issue with these fields missing from an OpenShift resource.

Reply:
ok, now using `_.get()` for `route spec` and `status`

Reply:
> k, now using _.get() for route spec and status

:(

I don't think we should...